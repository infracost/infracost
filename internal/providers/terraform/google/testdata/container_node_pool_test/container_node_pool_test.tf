provider "google" {
  credentials = "{\"type\":\"service_account\"}"
  region      = "us-central1"
}

resource "google_container_cluster" "default_regional" {
  name     = "default"
  location = "us-central1-a"
}

resource "google_container_cluster" "node_locations_regional" {
  name     = "node-locations"
  location = "us-central1-a"

  node_locations = [
    "us-central1-a",
    "us-central1-b",
  ]
}

resource "google_container_node_pool" "default_regional" {
  name    = "default"
  cluster = google_container_cluster.default_regional.id
}

resource "google_container_node_pool" "with_node_config_regional" {
  name       = "with-node-config"
  cluster    = google_container_cluster.default_regional.id
  node_count = 3

  node_config {
    machine_type    = "n1-standard-16"
    disk_size_gb    = 120
    disk_type       = "pd-ssd"
    local_ssd_count = 1

    guest_accelerator {
      type  = "nvidia-tesla-k80"
      count = 4
    }
  }
}

resource "google_container_node_pool" "with_custom_instance" {
  name       = "with-custom-instance"
  cluster    = google_container_cluster.default_regional.id
  node_count = 3

  node_config {
    machine_type = "n1-custom-6-20480"
  }
}

resource "google_container_node_pool" "with_guest_accelerator_a100" {
  name       = "with-a100"
  cluster    = google_container_cluster.default_regional.id
  node_count = 3

  node_config {
    machine_type    = "n1-standard-16"
    disk_size_gb    = 120
    disk_type       = "pd-ssd"
    local_ssd_count = 1

    guest_accelerator {
      type  = "nvidia-tesla-a100"
      count = 4
    }
  }
}

resource "google_container_node_pool" "cluster_node_locations_regional" {
  name       = "cluster-node-locations"
  cluster    = google_container_cluster.node_locations_regional.id
  node_count = 2
}

resource "google_container_node_pool" "node_locations_regional" {
  name       = "node-locations"
  cluster    = google_container_cluster.default_regional.id
  node_count = 2

  node_locations = [
    "us-central1-a",
    "us-central1-b",
  ]
}

resource "google_container_node_pool" "initial_node_count_regional" {
  name               = "initial-node-count"
  cluster            = google_container_cluster.default_regional.id
  initial_node_count = 4
}

resource "google_container_node_pool" "autoscaling_regional" {
  name    = "autoscaling"
  cluster = google_container_cluster.default_regional.id

  autoscaling {
    min_node_count = 2
    max_node_count = 10
  }
}

resource "google_container_cluster" "default_zonal" {
  name     = "default"
  location = "us-central1"
}

resource "google_container_cluster" "node_locations_zonal" {
  name     = "node-locations"
  location = "us-central1"

  node_locations = [
    "us-central1-a",
    "us-central1-b",
  ]
}

resource "google_container_node_pool" "default_zonal" {
  name    = "default"
  cluster = google_container_cluster.default_zonal.id
}

resource "google_container_node_pool" "cluster_node_locations_zonal" {
  name       = "cluster-node-locations"
  cluster    = google_container_cluster.node_locations_zonal.id
  node_count = 2
}

resource "google_container_node_pool" "node_locations_zonal" {
  name       = "node-locations"
  cluster    = google_container_cluster.default_zonal.id
  node_count = 2

  node_locations = [
    "us-central1-a",
    "us-central1-b",
  ]
}

resource "google_container_node_pool" "initial_node_count_zonal" {
  name               = "initial-node-count"
  cluster            = google_container_cluster.default_zonal.id
  initial_node_count = 4
}

resource "google_container_node_pool" "autoscaling_zonal" {
  name    = "autoscaling"
  cluster = google_container_cluster.default_zonal.id

  autoscaling {
    min_node_count = 2
    max_node_count = 10
  }
}

resource "google_container_cluster" "zonal_usage" {
  name     = "default"
  location = "us-central1-a"
}

resource "google_container_cluster" "regional_usage" {
  name     = "default"
  location = "us-central1"
}

resource "google_container_cluster" "node_locations_usage" {
  name     = "node-locations"
  location = "us-central1"

  node_locations = [
    "us-central1-a",
    "us-central1-b",
  ]
}

resource "google_container_node_pool" "zonal_usage" {
  name       = "zonal"
  cluster    = google_container_cluster.zonal_usage.id
  node_count = 3
}

resource "google_container_node_pool" "regional_usage" {
  name       = "regional"
  cluster    = google_container_cluster.regional_usage.id
  node_count = 3
}

resource "google_container_node_pool" "node_locations_usage" {
  name       = "node-locations"
  cluster    = google_container_cluster.node_locations_usage.id
  node_count = 3
}
