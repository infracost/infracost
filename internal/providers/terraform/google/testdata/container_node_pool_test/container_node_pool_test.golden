
 Name                                                        Monthly Qty  Unit   Monthly Cost 
                                                                                              
 google_container_cluster.default_regional                                                    
 ├─ Cluster management fee                                           730  hours        $73.00 
 └─ default_pool                                                                              
    ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)           2,190  hours        $73.38 
    └─ Standard provisioned storage (pd-standard)                    300  GB           $12.00 
                                                                                              
 google_container_cluster.default_zonal                                                       
 ├─ Cluster management fee                                           730  hours        $73.00 
 └─ default_pool                                                                              
    ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)           6,570  hours       $220.13 
    └─ Standard provisioned storage (pd-standard)                    900  GB           $36.00 
                                                                                              
 google_container_cluster.node_locations_regional                                             
 ├─ Cluster management fee                                           730  hours        $73.00 
 └─ default_pool                                                                              
    ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)           4,380  hours       $146.76 
    └─ Standard provisioned storage (pd-standard)                    600  GB           $24.00 
                                                                                              
 google_container_cluster.node_locations_usage                                                
 ├─ Cluster management fee                                           730  hours        $73.00 
 └─ default_pool                                                                              
    ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)           4,380  hours       $146.76 
    └─ Standard provisioned storage (pd-standard)                    600  GB           $24.00 
                                                                                              
 google_container_cluster.node_locations_zonal                                                
 ├─ Cluster management fee                                           730  hours        $73.00 
 └─ default_pool                                                                              
    ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)           4,380  hours       $146.76 
    └─ Standard provisioned storage (pd-standard)                    600  GB           $24.00 
                                                                                              
 google_container_cluster.regional_usage                                                      
 ├─ Cluster management fee                                           730  hours        $73.00 
 └─ default_pool                                                                              
    ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)           6,570  hours       $220.13 
    └─ Standard provisioned storage (pd-standard)                    900  GB           $36.00 
                                                                                              
 google_container_cluster.zonal_usage                                                         
 ├─ Cluster management fee                                           730  hours        $73.00 
 └─ default_pool                                                                              
    ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)           2,190  hours        $73.38 
    └─ Standard provisioned storage (pd-standard)                    300  GB           $12.00 
                                                                                              
 google_container_node_pool.autoscaling_regional                                              
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              1,460  hours        $48.92 
 └─ Standard provisioned storage (pd-standard)                       200  GB            $8.00 
                                                                                              
 google_container_node_pool.autoscaling_zonal                                                 
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              4,380  hours       $146.76 
 └─ Standard provisioned storage (pd-standard)                       600  GB           $24.00 
                                                                                              
 google_container_node_pool.cluster_node_locations_regional                                   
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              2,920  hours        $97.84 
 └─ Standard provisioned storage (pd-standard)                       400  GB           $16.00 
                                                                                              
 google_container_node_pool.cluster_node_locations_zonal                                      
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              2,920  hours        $97.84 
 └─ Standard provisioned storage (pd-standard)                       400  GB           $16.00 
                                                                                              
 google_container_node_pool.default_regional                                                  
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              2,190  hours        $73.38 
 └─ Standard provisioned storage (pd-standard)                       300  GB           $12.00 
                                                                                              
 google_container_node_pool.default_zonal                                                     
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              6,570  hours       $220.13 
 └─ Standard provisioned storage (pd-standard)                       900  GB           $36.00 
                                                                                              
 google_container_node_pool.initial_node_count_regional                                       
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              2,920  hours        $97.84 
 └─ Standard provisioned storage (pd-standard)                       400  GB           $16.00 
                                                                                              
 google_container_node_pool.initial_node_count_zonal                                          
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              8,760  hours       $293.51 
 └─ Standard provisioned storage (pd-standard)                     1,200  GB           $48.00 
                                                                                              
 google_container_node_pool.node_locations_regional                                           
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              2,920  hours        $97.84 
 └─ Standard provisioned storage (pd-standard)                       400  GB           $16.00 
                                                                                              
 google_container_node_pool.node_locations_usage                                              
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              5,840  hours       $195.67 
 └─ Standard provisioned storage (pd-standard)                       800  GB           $32.00 
                                                                                              
 google_container_node_pool.node_locations_zonal                                              
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              2,920  hours        $97.84 
 └─ Standard provisioned storage (pd-standard)                       400  GB           $16.00 
                                                                                              
 google_container_node_pool.regional_usage                                                    
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              8,760  hours       $293.51 
 └─ Standard provisioned storage (pd-standard)                     1,200  GB           $48.00 
                                                                                              
 google_container_node_pool.with_custom_instance                                              
 ├─ Custom instance CPU (Linux/UNIX, on-demand, N1 6 vCPUs)        2,190  hours       $305.13 
 ├─ Custom Instance RAM (Linux/UNIX, on-demand, N1 20 GB)          2,190  hours       $136.31 
 └─ Standard provisioned storage (pd-standard)                       300  GB           $12.00 
                                                                                              
 google_container_node_pool.with_guest_accelerator_a100                                       
 ├─ Instance usage (Linux/UNIX, on-demand, n1-standard-16)         2,190  hours     $1,165.07 
 ├─ SSD provisioned storage (pd-ssd)                                 360  GB           $61.20 
 ├─ Local SSD provisioned storage                                  1,125  GB           $90.00 
 └─ NVIDIA Tesla A100 (on-demand)                                  8,760  hours    $17,990.72 
                                                                                              
 google_container_node_pool.with_node_config_regional                                         
 ├─ Instance usage (Linux/UNIX, on-demand, n1-standard-16)         2,190  hours     $1,165.07 
 ├─ SSD provisioned storage (pd-ssd)                                 360  GB           $61.20 
 ├─ Local SSD provisioned storage                                  1,125  GB           $90.00 
 └─ NVIDIA Tesla K80 (on-demand)                                   8,760  hours     $2,759.40 
                                                                                              
 google_container_node_pool.zonal_usage                                                       
 ├─ Instance usage (Linux/UNIX, on-demand, e2-medium)              2,920  hours        $97.84 
 └─ Standard provisioned storage (pd-standard)                       400  GB           $16.00 
                                                                                              
 OVERALL TOTAL                                                                     $27,705.30 
──────────────────────────────────
23 cloud resources were detected:
∙ 23 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file